File: backend/internal/dto/response/transfer.go
package response

import "time"

type TransferOrderResp struct {
	ID              uint64    `json:"id"`
	FromAccountType string    `json:"from_account_type"`
	ToAccountType   string    `json:"to_account_type"`
	Currency        string    `json:"currency"`
	Amount          string    `json:"amount"`
	Status          string    `json:"status"`
	CreatedAt       time.Time `json:"created_at"`
}

type TransferOrderListResp struct {
	Orders []TransferOrderResp `json:"orders"`
	Total  int64               `json:"total"`
}

File: admin/src/types/transfer.ts
export interface AdminTransfer {
  id: number;
  platform_order_id: string;
  merchant_id: number;
  merchant_email: string;
  from_account_type: string;
  to_account_type: string;
  currency: string;
  amount: string;
  status: string;
  kun_order_id?: string;
  kun_request_no?: string;
  created_at: string;
}

export interface AdminTransferListResponse {
  transfers: AdminTransfer[];
  total: number;
}

export interface AdminTransferSyncResponse {
  order_id: number;
  status: string;
  kun_status: string;
  updated: boolean;
}
